### PR TITLE
Fix non-bridging settings

### PR DESCRIPTION
**Pull request by luis4a0**
The code in Daemon::set() did not take into account that Daemon::get_bridged_interface_name() could throw. Throwing is correct in all the other calls, so only the exception thrown in `set()` was catched this time.

### CODE REVIEW

**Review by ricab — CHANGES_REQUESTED** [review]
Hey @luis4a0, I think there are a few things you can do to enhance this (inline).

Off-topic: I also see problems with the whole user-authorization scheme (e.g. not thread-safe, leaks semantics of specific settings to `Daemon::set`). I would rather see `MP_SETTINGS` augmented with a `bool explicitly_authorized` param instead. I guess that ship has sailed for now though... Not this PR in any case.

File: src/daemon/daemon.cpp [hunk 1]
```diff
@@ -2438,18 +2438,29 @@ try
 
     auto key = request->key();
     auto val = request->val();
-    auto bridge_name = get_bridged_interface_name();
 
-    if (request->authorized())
+    std::string bridge_name;
+    try
+    {
+        bridge_name = get_bridged_interface_name(); // Throws if the interface name is empty.
```
[the remark this answers is not in the document]
Comment (luis4a0) [reply src/daemon/daemon.cpp]: Right, changed.

File: src/daemon/daemon.cpp [hunk 2]
```diff
@@ -2438,18 +2438,29 @@ try
...
+
+        if (request->authorized())
+        {
+            user_authorized_bridges.insert(bridge_name);
+        }
+    }
+    catch (const std::runtime_error&)
```
[the remark this answers is not in the document]
Comment (luis4a0) [reply src/daemon/daemon.cpp]: Yes, I addressed the previous comment so this is not relevant anymore.

File: src/daemon/daemon.cpp [hunk 3]
```diff
@@ -2438,9 +2438,9 @@ try
 
     auto key = request->key();
     auto val = request->val();
-    auto bridge_name = get_bridged_interface_name();
+    auto bridge_name = MP_SETTINGS.get(mp::bridged_interface_key).toStdString();
```
Comment (ricab) [inline src/daemon/daemon.cpp:2441]: IIUC, you should only need to read this if we are indeed dealing with authorized bridges. For any other set requests, this adds a pointless disk read. Therefore, can we read this only if the request is authorized? That would filter most cases.

Comment (ricab) [reply src/daemon/daemon.cpp:2441]: I understand you need the `bridge_name` var in this scope to later remove it (and you will still need it with the scope guard). But you can initialize it as empty by default, and only fetch the actual value when necessary.

Comment (luis4a0) [reply src/daemon/daemon.cpp]: Right, done.

**Comment by luis4a0** [comment]
Hey @ricab, thanks for the review! Yes, we can address the user-authorization stuff in a new PR.

File: src/daemon/daemon.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (ricab) [reply src/daemon/daemon.cpp]: OK yeah, that would be great!

File: src/daemon/daemon.cpp [hunk 4]
```diff
@@ -2449,7 +2451,12 @@ try
     MP_SETTINGS.set(QString::fromStdString(key), QString::fromStdString(val));
     mpl::log(mpl::Level::debug, category, fmt::format("Succeeded setting {}={}", key, val));
 
-    user_authorized_bridges.erase(bridge_name);
+    auto auth_guard = sg::make_scope_guard([this, &bridge_name]() noexcept {
+        mp::top_catch_all(category, [this, &bridge_name]() {
+            if (!bridge_name.empty())
+                user_authorized_bridges.erase(bridge_name);
+        });
+    });
```
[the remark this answers is not in the document]
Comment (luis4a0) [reply src/daemon/daemon.cpp:2455]: Ohhh, right. That's the purpose of the scope guard. Done.

**Pull request merged**